### PR TITLE
PR for Lesson010 Mailroom Functional Program

### PR DESCRIPTION
**Pull request by LouReis**
Hello,
Please review & grade the following for Lesson010:
mailroom_fp.py
test_mailroom_fp.py

Thanks & Regards
Lou

### CODE REVIEW

**Comment by LouReis** [comment]
Hello Natasha,

Also added update for Lesson09 mailroom5.py...it's now included in this PR also

I attempted to implement DonorCollection and do some refactoring per your comments but I'm a bit stuck and it's not working quite right...if you could please provide feedback.

Thanks & Regards,
Lou

File: students/LouReis/Lesson09/mailroom5.py [hunk 1]
```diff
@@ -113,12 +117,30 @@ def __eq__(self, other):
...
+            self.donors[name] = Donor(name, donation)
+    def get_report(self):
+        for donor_obj in self.donors.values():
+            donor_obj.total_donations
+            print (donor_obj.name, donor_obj.totoal_donations)
+            donation_report2()
+
+    def send_letters(self):
+        for donor_obj in self.donors.values():
+            donor_obj.get_thank_you_letter()
```
Comment (natasha-aleksandrova) [inline students/LouReis/Lesson09/mailroom5.py]: I think you want to print this right? `print(donor_obj.get_thank_you_letter())`

File: students/LouReis/Lesson09/mailroom5.py [hunk 2]
```diff
@@ -113,12 +117,30 @@ def __eq__(self, other):
...
+
+    def add_donation(self, name, donation):
+        if self.donors.get(name):
+            self.donors[name].add_donation(donation)
+        else:
+            self.donors[name] = Donor(name, donation)
+    def get_report(self):
+        for donor_obj in self.donors.values():
+            donor_obj.total_donations
+            print (donor_obj.name, donor_obj.totoal_donations)
```
Comment (natasha-aleksandrova) [inline students/LouReis/Lesson09/mailroom5.py]: use each donor object data to generate your report

**Comment by LouReis** [comment]
Ok...can you please review & grade my Lesson 10 exercise?  I’ve included it the last few pull requests and it’s been completed for several days.

Thanks 
Lou

Sent using the GMX mail app

**Comment by natasha-aleksandrova** [comment]
weird.. I didn't see your lesson 10 submission on your previous PR but I do see it now and you're good to go.

**Comment by LouReis** [comment]
Ok, thanks for taking care of that.

 

Can you mark my Python 210 as complete & post my final grade?  I believe that I have completed all of the work & that it has all been graded now (please let me know if I'm missing anything).  I'd like this class shown as complete ASAP so that I can begin the next one. 

 

I will work on improving a few of my scripts over the next few days and will submit them to git, this shouldn't have any affect on my grade though.  I want to make sure I improve & understand the OO/classes scripting...


Thanks,
Lou

 

Sent: Friday, January 18, 2019 at 4:05 PM

**Pull request merged**